auacm/angular/src/app/pipes/search-filter.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+import { Problem } from '../models/problem';
+
+@Pipe({
+  name: 'searchFilter'
+})
+export class SearchFilterPipe implements PipeTransform {
+
+  transform(array: Problem[], search: string): any {
+    if (search === "") {return array};
+    var validMatches = [];
+    for (var i = 0; i < array.length; i++) {
+      if (array[i].name.toLowerCase().startsWith(search.toLowerCase()) 
```

Comment:
Consider making a local variable for `search.toLowerCase()` rather than calling it repeatedly